### PR TITLE
[bun:sqlite] Support multiple statements in db.run()

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

Fixes https://github.com/oven-sh/bun/issues/5078

### How did you verify your code works?

There is a test

### CODE REVIEW

File: src/bun.js/bindings/sqlite/JSSQLStatement.cpp [hunk 1]
```diff
@@ -945,65 +960,92 @@ JSC_DEFINE_HOST_FUNCTION(jsSQLStatementExecuteFunction, (JSC::JSGlobalObject * l
...
-            if (UNLIKELY(!reb.isNumber())) {
-                sqlite3_finalize(statement);
-                return JSValue::encode(reb); /* this means an error */
+    while (sqlStringHead && sqlStringHead < end) {
+        if (UNLIKELY(isSkippedInSQLiteQuery(*sqlStringHead))) {
+            sqlStringHead++;
+
+            while (sqlStringHead < end && isSkippedInSQLiteQuery(*sqlStringHead))
+                sqlStringHead++;
+        }
```
Comment (gvilums) [inline src/bun.js/bindings/sqlite/JSSQLStatement.cpp:996]: couldn't this just be the `while` loop?

File: src/bun.js/bindings/sqlite/JSSQLStatement.cpp [hunk 2]
```diff
@@ -945,65 +960,92 @@ JSC_DEFINE_HOST_FUNCTION(jsSQLStatementExecuteFunction, (JSC::JSGlobalObject * l
...
+
+        AutoDestructingSQLiteStatement sql;
+        const char* tail = nullptr;
+
+        // Bounds checks
+        ASSERT(end >= sqlStringHead);
+        ASSERT(end - sqlStringHead >= 0);
```
Comment (gvilums) [inline src/bun.js/bindings/sqlite/JSSQLStatement.cpp:1003]: these two are redundant

**Review by gvilums — APPROVED** [review]

File: src/bun.js/bindings/sqlite/JSSQLStatement.cpp [hunk 3]
```diff
@@ -899,6 +909,11 @@ JSC_DEFINE_HOST_FUNCTION(jsSQLStatementLoadExtensionFunction, (JSC::JSGlobalObje
     RELEASE_AND_RETURN(scope, JSValue::encode(JSC::jsUndefined()));
 }
 
+static bool isSkippedInSQLiteQuery(const char c)
+{
+    return c == ' ' || c == ';' || (c >= '\t' && c <= '\r');
```
Comment (jdalton) [inline src/bun.js/bindings/sqlite/JSSQLStatement.cpp:914]: Is the `c >= '\t`...` looking for line terminators as defined by ECMA262?:
https://tc39.es/ecma262/#prod-LineTerminator 

Includes also U+2028 (LINE SEPARATOR) and U+2029 (PARAGRAPH SEPARATOR).

Comment (Jarred-Sumner) [reply src/bun.js/bindings/sqlite/JSSQLStatement.cpp:914]: these are specific to sqlite, not JS

File: src/bun.js/bindings/sqlite/JSSQLStatement.cpp (hunk 2 above)
Comment (Jarred-Sumner) [reply src/bun.js/bindings/sqlite/JSSQLStatement.cpp:1003]: yeah

**Pull request merged**